### PR TITLE
Refactor resource loading in Runtime

### PR DESCRIPTION
**Pull request by D8H**
### Changes
- Use async/await
- Pre-filter assets by types (it's iterated only once currently, but it will be useful for loading per layout)
- Use a Map for resources because loading per layout will need an access by resource names.

### CODE REVIEW

**Comment by D8H** [comment]
### Manual tests

* Tile map

image

* Bitmap text

image

File: GDJS/Runtime/pixi-renderers/pixi-image-manager.ts [hunk 1]
```diff
@@ -310,16 +310,15 @@ namespace gdjs {
      * @param onProgress Callback called each time a new file is loaded.
      * @param onComplete Callback called when loading is done.
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/pixi-image-manager.ts:311]: Remove `onComplete` parameter documentation here and in other managers

File: GDJS/Runtime/pixi-renderers/pixi-bitmapfont-manager.ts [hunk 2]
```diff
@@ -253,50 +259,48 @@ namespace gdjs {
...
-                  error
-              );
-            })
-            .then(() => {
-              loadedCount++;
-              onProgress(loadedCount, bitmapFontResources.length);
-            });
+      await Promise.all(
+        gdjs.mapIterable(preloadedResources, async (bitmapFontResource) => {
+          try {
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/pixi-bitmapfont-manager.ts:275]: This is an array right? Can you do a map?

```suggestion
      await Promise.all(
        preloadedResources.map(async (bitmapFontResource) => {
          try {
```

File: GDJS/Runtime/jsonmanager.ts [hunk 3]
```diff
@@ -60,40 +65,42 @@ namespace gdjs {
...
-      let loaded = 0;
+    async preloadJsons(
+      onProgress: (loadedCount: integer, totalCount: integer) => void
+    ): Promise<integer> {
+      const preloadedResources = [
+        ...gdjs.filterIterable(
+          this._resources.values(),
+          (resource) => !resource.disablePreload
+        ),
+      ];
```
Comment (4ian) [inline GDJS/Runtime/jsonmanager.ts:77]: Could you do?

```suggestion
      const preloadedResources = [...this._resources.values()](
          .filter((resource) => !resource.disablePreload);
```

**Review by 4ian — COMMENTED** [review]
Would lgtm if we could avoid the two new global functions

File: GDJS/Runtime/jsonmanager.ts (hunk 3 above)
Comment (D8H) [reply GDJS/Runtime/jsonmanager.ts]: Maybe I micro optimized it.

Comment (4ian) [reply GDJS/Runtime/jsonmanager.ts]: Thanks for adapting this 👍 Let's keep things with simple and stupid arrays/maps :)

**Review by 4ian — APPROVED** [review]
More deletions than added changes, I like that!
image

**Pull request merged**